hooks/useRailgunProvider.tsx

```diff
@@ -1,19 +1,43 @@
 import { useEffect, useState } from 'react';
 import { setProviderForNetwork } from '@railgun-community/quickstart';
+import { BigNumber } from 'ethers';
 import { useNetwork, useProvider } from 'wagmi';
 import { networks } from '@/utils/networks';
 import { loadProviders } from '@/utils/railgun';
 
+// Fee is in bips, e.g. a value of 25 is a 0.25% fee.
+interface ShieldFee {
+  [chainId: number]: BigNumber;
+}
+
+const initialShieldFees: ShieldFee = {};
+Object.keys(networks).forEach((chainId) => {
+  // Current fees are 0.25% everywhere, so we initialize with that as a default -- If something goes
+  // wrong fetching fees, better to show a higher fee than a zero default fee.
+  initialShieldFees[Number(chainId)] = BigNumber.from('25');
+});
+
 export const useRailgunProvider = () => {
   const [isProviderLoaded, setProviderLoaded] = useState<Boolean>(false);
+  const [shieldFees, setShieldFees] = useState<ShieldFee>(initialShieldFees);
   const { chain } = useNetwork();
   const network = networks[chain?.id || 1];
   const provider = useProvider();
 
   useEffect(() => {
     const fn = async () => {
       setProviderLoaded(false);
-      await loadProviders();
+      const res = await loadProviders();
+
+      // Set the shield fees for each network. For convenience, we just update the
+      // `initialShieldFees` object then set it.
+      res.forEach((r) => {
```

Comment:
style nit: don't love mutating the `initialShieldFees` object.

```js
const shieldFeesFromNetwork = res.reduce((acc, response) => {
        const newFee = response.providerInfo.feesSerialized?.shield; // Personally think we should error loudly here if `feesSerialized` not available
        return {
           ...acc,
           [response.chainId]: BigNumber.from(newFee || initialShieldFees[response.chainId])
        }
}, {})

setShieldFees(shieldFeesFromNetwork)
```

Reply:
refactored in 95365db

> // Personally think we should error loudly here if `feesSerialized` not available

If `loadProvider` returns successfully this will never be not available (just asked railgun) so the right spot to handle this is in `railgun.ts`. We weren't handling the error case there originally, so worth pulling out into a separate issue? (related to https://github.com/ScopeLift/railgun-shielder/pull/78#discussion_r1092323102)

Reply:
tracking in #85, thanks!